internal/sdk/retry_client.go

```diff
@@ -23,6 +25,7 @@ func NewRetryClient(inner HTTPClient, maxRetries int) HTTPClient {
 	return &RetryClient{
 		inner:      inner,
 		maxRetries: maxRetries,
+		rand:       rand.New(rand.NewSource(0)),
```

Comment:
This generates deterministic behavior, which makes testing easier. I don't think it has significant implications for production code.

Reply:
If an app (like yours) creates 100,000 clients and they all have the same random source, and are all started at around the same instant, wouldn't that defeat the purpose of your implementation of the jittery backoff? 

Maybe it would be better to provide the `rand.Source` via a constructor parameter. So in the tests we pass `rand.NewSource(0)` to achieve deterministic/testable behavior and in the actual client builder (wireup) code we provide `rand.NewSource(time.Now().Unix())` or something else more suitable.